canonical-web-and-design/www.ubuntu.com
# "Releases" global template context

This tidies up the releases variables, keeping them in one easy-to-read file called `releases.yaml`.

This is to help with https://github.com/canonical-web-and-design/www.ubuntu.com/pull/5190.

In that PR, @pmahnke could simply add a `checksums:` key to `releases.yaml`:

``` yaml
checksums:
  desktop:
    18.04.2: 22580b9f3b186cc66818e60f44c46f795d708a1ad86b9225c458413b638459c4
    19.04: 2da6f8b5c65b71b040c5c510311eae1798545b8ba801c9b63e9e3fd3c0457cbe
```

And then these checksums could be accessed in templates with:

```
{{ releases.checksums.desktop | keyvalue : releases.latest.full_version }}
```

QA
--

`./run`, go to e.g. `/download/desktop`, check the numbers look right.

Now change something in `releases.yaml` and check the value changed.

For extra marks, check all the changed templates, to check they don't cause and error, or fail to show information correctly.
